filebeat/spooler/spooler.go

```diff
@@ -61,24 +59,32 @@ func (s *Spooler) Start() {
 func (s *Spooler) run() {
 	defer s.wg.Done()
 
-	ticker := time.NewTicker(s.config.idleTimeout / 2)
-
 	logp.Info("Starting spooler: spool_size: %v; idle_timeout: %s",
 		s.config.spoolSize, s.config.idleTimeout)
 
+	timer := time.NewTimer(s.config.idleTimeout)
+
 loop:
 	for {
 		select {
 		case <-s.exit:
-			ticker.Stop()
 			break loop
 		case event := <-s.Channel:
 			if event != nil {
 				s.queue(event)
 			}
-		case <-ticker.C:
-			s.timedFlush()
+
+			// Stop timer and drain channel. See https://golang.org/pkg/time/#Timer.Reset
+			if !timer.Stop() {
+				<-timer.C
+			}
```

Comment:
the timer shall be stopped only when queue is flushed, not per event. e.g. `queue` can return a `bool` indicating if queue has been flushed.